File: backend/ZigBackend/src/main/java/com/example/zigbackend/dto/ZahtevZaPriznanjeZigaDTO.java
package com.example.zigbackend.dto;

import com.example.zigbackend.model.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlElementWrapper;
import javax.xml.bind.annotation.XmlRootElement;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@XmlRootElement
public class ZahtevZaPriznanjeZigaDTO {
    private LiceDTO podnosilacPrijave;
    private LiceDTO punomocnik;
    private LiceDTO predstavnik;
//    public Zig zig;
    public ZigDTO zigDTO;
//    @XmlElementWrapper(name = "klasaDTO")
//    @XmlElement(name = "klasa")
//    private List<Klasa> klasa;
    private String klasaConcatenated;
    private EZatrazeno_pravo_prvenstva_i_osnov zatrazenoPravoPrvenstvaIOsnov;
//    @XmlElementWrapper(name = "neededPrilogsDTO")
//    @XmlElement(name = "neededPrilogs")
//    private List<String> neededPrilogs;
    private String neededPrilogsConcatenated;
    private String statusPrilogPunomocje;
//    private NeededPrilogsDTO neededPrilogsDTO;

    // does not have Taksa, Broj_prijave_ziga, Datum_podnosenja and Status
}

File: frontend/src/app/model/zigModels/ZahtevZaPriznanjeZigaDTO.ts
import {LiceZahtevaZig} from "./LiceZahtevaZig";
import {Zig} from "./Zig";
import {Klasa} from "./Klasa";
import {ZigDTO} from "./ZigDTO";

export class BojaDTO {
  boja: string[] = [];

  public BojaDTO(){
    this.boja = [];
  }
}

export class KlasaDTO {
  klasa: Klasa[] = [];

  public KlasaDTO(){
    this.klasa = [];
  }
}

export class NeededPrilogsDTO {
  neededPrilogs: string[] = [];

  public NeededPrilogsDTO(){
    this.neededPrilogs = [];
  }
}

export class ZahtevZaPriznanjeZigaDTO {
  podnosilacPrijave: LiceZahtevaZig = new LiceZahtevaZig();
  punomocnik: LiceZahtevaZig = new LiceZahtevaZig();
  predstavnik: LiceZahtevaZig = new LiceZahtevaZig();
  // zig: Zig = new Zig();
  // klasa: Klasa[] = [];
  zatrazenoPravoPrvenstvaIOsnov: string = "";
  // neededPrilogs: string[] = [];
  statusPrilogPunomocje: string = "";
  zigDTO: ZigDTO = new ZigDTO();
  // bojaDTO: BojaDTO = new BojaDTO;
  // klasaDTO: KlasaDTO = new KlasaDTO;
  // neededPrilogsDTO: NeededPrilogsDTO = new NeededPrilogsDTO;
  klasaConcatenated: string = "";
  neededPrilogsConcatenated: string = "";

  public ZahtevZaPriznanjeZigaDTO(){
    this.podnosilacPrijave = new LiceZahtevaZig();
    this.punomocnik = new LiceZahtevaZig();
    this.predstavnik = new LiceZahtevaZig();
    // this.zig = new Zig();
    // this.klasa = [];
    this.zatrazenoPravoPrvenstvaIOsnov = "";
    // this.neededPrilogs = [];
    this.statusPrilogPunomocje = "";
    this.zigDTO= new ZigDTO();
    // this.bojaDTO = new BojaDTO;
    // this.klasaDTO = new KlasaDTO;
    // this.neededPrilogsDTO = new NeededPrilogsDTO;
    this.klasaConcatenated = "";
    this.neededPrilogsConcatenated = "";
  }
}
